chromadb/types.py

```diff
@@ -48,21 +40,19 @@ class Segment(TypedDict):
     metadata: Optional[Metadata]
 
 
-S = TypeVar("S", bound="SeqId")
-
+# The desire here is for SeqID to be any type that can be compared to other values of
+# the same type to establish a linear order.
 
-class SeqId(Protocol):
-    def __eq__(self, other: Any) -> bool:
-        ...
+# This is surprisingly difficult to express in Python. ints, for example, do not
+# "support" __eq__ and __lt__ so using a protocol won't work.
+SeqId = Any
```

Comment:
Can we instead just strongly type it to what we expect - int, float, our SeqID class from pulsar?